contracts/aux/auto-alex-v3.clar

```diff
@@ -0,0 +1,179 @@
+	
+(define-public (burn-fixed (amount uint) (sender principal))
+	(burn amount sender))
+
+(define-public (burn-many (senders (list 200 {amount: uint, sender: principal})))
+	(fold check-err (map burn-many-iter senders) (ok true)))
+
+(define-public (burn-fixed-many (senders (list 200 {amount: uint, sender: principal})))
+	(burn-many senders))
+
+;; read-only functions
+
+(define-read-only (is-dao-or-extension)
+	(ok (asserts! (or (is-eq tx-sender .lisa-dao) (contract-call? .lisa-dao is-extension contract-caller)) err-unauthorised)))
+
+(define-read-only (get-name)
+	(ok (var-get token-name)))
+
+(define-read-only (get-symbol)
+	(ok (var-get token-symbol)))
+
+(define-read-only (get-token-uri)
+	(ok (var-get token-uri)))
+
+(define-read-only (get-decimals)
+	(ok token-decimals))
+
+(define-read-only (get-balance (who principal))
+	(ok (get-shares-to-tokens (unwrap-panic (get-share who)))))
+
+(define-read-only (get-balance-fixed (who principal))
+	(get-balance who))
+
+(define-read-only (get-total-supply)
+	(get-reserve))
+
+(define-read-only (get-total-supply-fixed)
+	(get-total-supply))
+
+(define-read-only (get-share (who principal))
+	(ok (ft-get-balance auto-alex-v3 who)))
+
+(define-read-only (get-share-fixed (who principal))
+	(get-share who))
+
+(define-read-only (get-total-shares)
+	(ok (ft-get-supply auto-alex-v3)))
+
+(define-read-only (get-total-shares-fixed)
+	(get-total-shares))
+
+(define-read-only (get-reserve)
+	(ok (var-get reserve)))
+
+(define-read-only (get-tokens-to-shares (amount uint))
+	(if (is-eq (get-reserve) (ok u0))
+		amount
+		(/ (* amount (unwrap-panic (get-total-shares))) (unwrap-panic (get-reserve)))))
+
+(define-read-only (get-shares-to-tokens (shares uint))
+	(if (is-eq (get-total-shares) (ok u0))
+		shares
+		(/ (* shares (unwrap-panic (get-reserve))) (unwrap-panic (get-total-shares)))))
+
+;; public calls
+
+(define-public (transfer (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
+	(let (
+			(shares (get-tokens-to-shares amount)))
+		(asserts! (or (is-eq tx-sender sender) (is-eq contract-caller sender)) err-unauthorised)
+		(try! (ft-transfer? auto-alex-v3 shares sender recipient))
+		(match memo to-print (print to-print) 0x)
+		(print { notification: "transfer", payload: { amount: amount, shares: shares, sender: sender, recipient: recipient } })
+		(ok true)))
+
+(define-public (transfer-fixed (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
+	(transfer amount sender recipient memo))
+
+;; private functions
+
+(define-private (burn-many-iter (item {amount: uint, sender: principal}))
+	(burn (get amount item) (get sender item)))
+
+(define-private (check-err (result (response bool uint)) (prior (response bool uint)))
+    (match prior ok-value result err-value (err err-value)))
+
+;; staking related fuctions
+
+(use-trait ft-trait 'SP102V8P0F7JX67ARQ77WEA3D3CFB5XW39REDT0AM.trait-sip-010.sip-010-trait)
```

Comment:
This is not sip10, but sip10 + `-fixed` methods. Maybe add a comment

Reply:
https://github.com/lisalab-io/liquid-stacking/pull/56/commits/20107362b60f7890c0538305e7089746f26b6ac3